assets/js/base/hooks/use-url-query-string.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useReducer } from '@wordpress/element';
+import { addQueryArgs, getQueryArg } from '@wordpress/url';
+import { useQueryStateContext } from '@woocommerce/base-context/query-state-context';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+
+const hasWindowDependencies =
+	typeof window === 'object' &&
+	window.hasOwnProperty( 'history' ) &&
+	window.hasOwnProperty( 'location' ) &&
+	typeof window.addEventListener === 'function' &&
+	typeof window.removeEventListener === 'function';
+
+const update = ( urlKey, urlValue ) => {
```

Comment:
I'm curious about our conventions for naming actions - these seem a little terse to me, personally I'd suffix with `Action` or put them in an `actions = { update:, reset: }` object, so it's clearer what these are when they are used. Happy with this terse form if that's our standard :) 

Also curious about putting the action type in a constant rather than literal. Probably cleaner as is since it's contained within this file.